internal/cli/serverless/sqluser/create.go

```diff
@@ -0,0 +1,302 @@
+				}
+
+				// interactive mode
+				project, err := cloud.GetSelectedProject(ctx, h.QueryPageSize, d)
+				if err != nil {
+					return err
+				}
+				projectID := project.ID
+
+				cluster, err := cloud.GetSelectedCluster(ctx, projectID, h.QueryPageSize, d)
+				if err != nil {
+					return err
+				}
+				clusterID = cluster.ID
+				userPrefix, err = GetUserPrefix(ctx, d, clusterID)
+				if err != nil {
+					return errors.Trace(err)
+				}
+
+				userRole, err = cloud.GetSelectedBuiltinRole()
+				if err != nil {
+					return err
+				}
+
+				// variables for input
+				fmt.Fprintln(h.IOStreams.Out, color.BlueString("Please input the following options"))
+
+				p := tea.NewProgram(initialCreateInputModel(userPrefix))
+				inputModel, err := p.Run()
+				if err != nil {
+					return errors.Trace(err)
+				}
+				if inputModel.(ui.TextInputModel).Interrupted {
+					return util.InterruptError
+				}
+
+				userName = inputModel.(ui.TextInputModel).Inputs[createSQLUserField[flag.UserName]].Value()
+				password = inputModel.(ui.TextInputModel).Inputs[createSQLUserField[flag.Password]].Value()
+
+			} else {
+				// non-interactive mode doesn't need projectID
+				cID, err := cmd.Flags().GetString(flag.ClusterID)
+				if err != nil {
+					return errors.Trace(err)
+				}
+				clusterID = cID
+
+				userPrefix, err = GetUserPrefix(ctx, d, clusterID)
+				if err != nil {
+					return errors.Trace(err)
+				}
+
+				uName, err := cmd.Flags().GetString(flag.UserName)
+				if err != nil {
+					return errors.Trace(err)
+				}
+				userName = uName
+
+				pw, err := cmd.Flags().GetString(flag.Password)
+				if err != nil {
+					return errors.Trace(err)
+				}
+				password = pw
+
+				uRole, err := cmd.Flags().GetString(flag.UserRole)
+				if err != nil {
+					return errors.Trace(err)
+				}
+				userRole = uRole
+			}
+
+			// generate the built-in role
+			builtinRole, customRoles := GetBuiltinRoleAndCustomRoles(userRole)
+
+			params := iamApi.NewPostV1beta1ClustersClusterIDSQLUsersParams().
+				WithClusterID(clusterID).
+				WithSQLUser(&iamModel.APICreateSQLUserReq{
+					AuthMethod:  util.MYSQLNATIVEPASSWORD,
+					UserName:    userName,
+					Password:    password,
+					BuiltinRole: builtinRole,
+					CustomRoles: customRoles,
+				}).
+				WithContext(ctx)
+
+			_, err = d.CreateSQLUser(params)
+			if err != nil {
+				return errors.Trace(err)
+			}
+
+			_, err = fmt.Fprintln(h.IOStreams.Out, color.GreenString("SQL user %s.%s is created", userPrefix, userName))
+			if err != nil {
+				return err
+			}
+			return nil
+
+		},
+	}
+
+	CreateCmd.Flags().StringP(flag.ClusterID, flag.ClusterIDShort, "", "The ID of the cluster.")
+	CreateCmd.Flags().StringP(flag.UserName, "", "", "The name of the SQL user.")
+	CreateCmd.Flags().StringP(flag.Password, "", "", "The password of the SQL user.")
+	CreateCmd.Flags().StringP(flag.UserRole, "", "", "The built-in role of the SQL user.")
```

Comment:
It should provide valid values for the user, like https://github.com/tidbcloud/tidbcloud-cli/blob/main/internal/cli/serverless/dataimport/start/start.go#L150

Reply:
> It should provide valid values for the user, like https://github.com/tidbcloud/tidbcloud-cli/blob/main/internal/cli/serverless/dataimport/start/start.go#L150

Do you mean the `UserRole`? Users are free to choose the content they wish to fill in